src/openforms/submissions/views.py

```diff
@@ -235,3 +240,41 @@ def get_sendfile_opts(self) -> dict:
             "mimetype": submission_attachment.content_type,
         }
         return opts
+
+
+class SearchSubmissionForCosignFormView(PermissionRequiredMixin, FormView):
+    form_class = SearchSubmissionForCosignForm
+    template_name = "submissions/find_submission_for_cosign.html"
+    raise_exception = True
+
+    cosign_url = None
+
+    def has_permission(self):
+        """
+        The user should have authenticated with the auth plugin specified on the form for the co-sign component
+        """
+        form = get_object_or_404(Form, slug=self.kwargs["form_slug"])
+        cosign_component = form.get_cosign_component()
+        expected_auth_plugin = cosign_component["authPlugin"]
+        return is_authenticated_with_plugin(self.request, expected_auth_plugin)
+
+    def get_initial(self):
+        return {"form_slug": self.kwargs["form_slug"]}
```

Comment:
probably makes more sense to:

1. Look up the form already with `get_object_or_404`
2. Pass the form instance to the form via `def get_form_kwargs()`
3. Extend the form `__init__` to take the `form` instance and store it in `self.form`. That way, you don't need the form slug hidden input at all, you just get the form from the URL and perform the access control checks.